Attempt of module 10

Jiymeade opened the pull request:
Minimal Requirements

Adjust ServerPlayerMovement's movement and add fire controls talking to bullet spawner Bullet Spawner system attached to serverPlayerPrefab Projectile and handling collision
Health System with simple image slider to update and sync Pickup example using ammo and a simple spawner in the scene Very basic Game Logic

Jiymeade committed:
Attempt of module 10

Minimal Requirements

Adjust ServerPlayerMovement's movement and add fire controls talking to bullet spawner
Bullet Spawner system attached to serverPlayerPrefab
Projectile and handling collision
Health System with simple image slider to update and sync
Pickup example using ammo and a simple spawner in the scene
Very basic Game Logic

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/Player/ServerPlayerMovements.cs:
```diff
@@ -3,16 +3,20 @@
+    [SerializeField] private BulletSpawnerScript _bulletSpawner;
     public CharacterController _CC;
     private MyPlayerInputActions _playerInput;
+    private static readonly int IsSprinting = Animator.StringToHash(name: "IsSprinting");
```
There was Hash Errors, this  StringToHash is case sensitive. Ensure that the animator's parameters are the same. it's currently has a lowercase i.

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/Player/ServerPlayerMovements.cs:
```diff
@@ -49,31 +53,40 @@ void FixedUpdate()
         if(isPunching){_myNetAnimator.SetTrigger("PunchTrigger"); }
 
-       _myAnimator.SetBool(name: "isSprinting", isSprinting);
+       _myAnimator.SetBool(id:IsSprinting, isSprinting);
```
`_myAnimator.SetBool(id:IsSprinting, isSprinting);`  needs the 'id:" to be removed and changed to `_myAnimator.SetBool(IsSprinting, isSprinting);`

same thing for any other inputs that have id:  or name:. I think its because you saw some of my intellesense that hints what type of variable it comes from

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/Player/ServerPlayerMovements.cs:
```diff
@@ -49,31 +53,40 @@ void FixedUpdate()
         {
 
-            _CC.Move(_moveDirection * (_pSpeed * 1.3f) * Time.deltaTime);
+            _CC.Move(motion:_moveDirection * (_pSpeed * 1.3f * Time.deltaTime));
```
Similar as above, change both server and client side of _CC.Move
` _CC.Move(motion:_moveDirection * (_pSpeed * 1.3f * Time.deltaTime));`

to 

` _CC.Move(_moveDirection * (_pSpeed * 1.3f * Time.deltaTime));`

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/ProjectileObjs.cs:
```diff
@@ -0,0 +1,42 @@
+        [SerializeField] private float _damage = 10;
+        [SerializeField] private float destructTime = 5f;
+
+        public override void OnNetworkSpawn()
```
another intellesense misread, remove the routine: from` StartCoroutine(routine: AutoDestruct());`

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/TntScript.cs:
```diff
@@ -0,0 +1,46 @@
+        {
+            NetworkObject explosive = NetworkManager.Instantiate(ExplosionPrefab, transform.position, transform.rotation);
+
+            ExplosionPrefab.Spawn(destroyWithScene: true);
```
intellesense misread / typo remove destroyWithScene:  from `ExplosionPrefab.Spawn(destroyWithScene: true);`  also instead of ExplosionPrefab, you need to reference the new instantiated Network Object explosive... so the final result should be `explosive.Spawn(true);`

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/ProjectileObjs.cs:
```diff
@@ -0,0 +1,42 @@
+
+        private void OnCollisionEnter(Collision other)
+        {
+            if (other.gameObject.tag.Equals("Player") && other.gameObject.GetComponent<NetworkObject>().OwnerClientId != this.OwnerClientId) ;
```
there's an extra` ;` at the end of if `(other.gameObject.tag.Equals("Player") && other.gameObject.GetComponent<NetworkObject>().OwnerClientId != this.OwnerClientId) ;` you need to remove the ; in order to properly get the results of the if condition

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/ProjectileObjs.cs:
```diff
@@ -0,0 +1,42 @@
+        {
+            base.OnNetworkSpawn();
+            GetComponent<Rigidbody>().velocity = this.transform.forward * _speed;
+            StartCoroutine(routine: AutoDestruct());
```
` StartCoroutine(routine: AutoDestruct()); ` needs reoutine removed and it needs to be either StartCoroutine("AutoDestruct"); or

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/TntScript.cs:
```diff
@@ -0,0 +1,46 @@
+            base.OnNetworkSpawn();
+            if (IsServer)
+            {
+                StartCoroutine(methodName: "TriggerFuse");
```
typo / intellesense error remove the `methodName:` from this StartCoroutine

MRN-DX commented on UC NetCode Fall 2024/Assets/App/Resource/Scripts/TntScript.cs:
```diff
@@ -0,0 +1,46 @@
+
+        private IEnumerator TriggerFuse()
+        {
+            Debug.Log(message: "Fuse List");
```
remove `message:`

MRN-DX reviewed (changes requested):
I noticed you lost some of your lobby stuff, but the biggest thing is getting those physics to sync up.  You are more than half way there, where a lot of your troubles is just making sure your player prefabs have their respective scripts in the right place and the scripts with prefab references, target the asset prefabs, not the ones that are in the scene. 

There was some misreading of my IDE showing some intellesense items that you've actually written in, so you'll see a bunch of those comments.  There's also a few bugs I saw for you to make note of. This should be a good amount to get your started with the new changes. we can move along the way.

I

-Ensure to Tag "Player" in your prefabs
-your spawning TNT / ammo game object needs better position, placed over the play area.
-Physics can be tricky to play with. I find that changing the GroundPlane into a cube helps with some collision, just make sure to add rigi body, turn off gravity, and set constraints.
-Need to add rigibody to ServerPlayer, might want to constrain the X and Z rotations while you there. Increase Drag and mass. This unity plays with its physics
-Need to tweak rigi bodies collision types to be continuous for players, continuous Dynamic for bullets and pickup. This can help with accuracy.
![image](https://github.com/user-attachments/assets/72c1666f-f31c-41a4-a7bc-e1afd388c0c7)

If you need to do a one on one trouble shooting let me know

Jiymeade committed:
Changes

-Module 10 changed made based off of feedback.. There is still issues suck as the losing text not showing up and a repeated use of explosion also the character floating.
